microcosm_pubsub/matchers.py

```diff
@@ -0,0 +1,95 @@
+from dataclasses import dataclass
+from json import loads
+from typing import List, Optional, Mapping
+
+from hamcrest import not_none
+from hamcrest.library.collection.issequence_containinginorder import IsSequenceContainingInOrder
+from hamcrest.library.object.hasproperty import IsObjectWithProperty
+from hamcrest.core.helpers.wrap_matcher import wrap_matcher
+
+
+@dataclass
+class PublishedMessage:
+    """
+    A published message.
+
+    """
+    topic: str
+    message: Mapping[str, str]
+
+    @property
+    def media_type(self) -> Optional[str]:
+        return self.message.get("mediaType")
+
+    @property
+    def uri(self) -> Optional[str]:
+        return self.message.get("uri")
+
+    @classmethod
+    def from_call(cls, call) -> "PublishedMessage":
+        name, args, kwargs = call
+        topic = kwargs["TopicArn"]
```

Comment:
The right thing to do here is to leverage the enveloping code. I'm not sure if we need to since we are very unlikely to run a unit test with these matchers and also change the default enveloping.